### ISSUE TITLE
Inconsistency between the paper and the implementation

### ISSUE THREAD

**Issue by naoto0804** [vinthony/ghost-free-shadow-removal#4]
I'm trying to re-implement your model on PyTorch and I have a few questions about details.

- Feature aggregation
In Fig. 2, `netaggi_0` / `netaggm_0` seems to be concatenated. In the [code](https://github.com/vinthony/ghost-free-shadow-removal/blob/master/networks.py#L171-L173), it is not concatenated. Which should I follow? 

- Missing 1x1 conv?
In Fig. 2, 1x1 conv. layer is inserted before SPP module. In the [code](https://github.com/vinthony/ghost-free-shadow-removal/blob/master/networks.py#L184), it's not. Should I add it?

**Comment by vinthony** [comment]
Thanks for your useful help.

You can follow the model from the code. The results are based on the model in this repo.

It seems that there are some small mistakes in the figure although it might not highly influence the performance.

I will try to retrain the model based on the model from paper.
and add some Introduction in the README about these inconsistencies.

Thanks a lot!

**Comment by naoto0804** [comment]
That can happen when one tries a lot of configurations; Thank you for your prompt response, and I'm looking forward to see the result, thank you!

**Comment by naoto0804** [comment]
Sorry to ask you one more;

Can you elaborate a bit more about why you divide the losses by some magic numbers? 
([here](https://github.com/vinthony/ghost-free-shadow-removal/blob/master/utils.py#L120-L124))

**Comment by vinthony** [comment]
These numbers are following the setting of the framework of our baseline code in reflection removal [here](https://github.com/ceciliavision/perceptual-reflection-removal/blob/8df2c5a0f7e934e9689db01b23a73568f1acaa97/main.py#L185)

**Comment by vinthony** [comment]
@naoto0804 
I re-train the model with the model in the paper (Fig.2) on the SRD dataset without data augmentation for a quick experiment.
The performance improves a bit on the Non-Shadowed Region (NS in the following Table), while the performance in the shadowed region is the same. However, the setting from the paper uses more parameters. Here is the RMSE report:

Method | S |  NS | ALL
---- | --- | --- | ---
Figure.2 | 8.94 | **4.40** | **5.35**
This repo  | 8.94  | 4.80 |  5.67

So  I suggest you use the model from the paper.

**Comment by naoto0804** [comment]
Wow, thank you so much!

**Comment by naoto0804** [comment]
Why do you initialize almost all the convolutional layers by [identity_initializer](https://github.com/vinthony/ghost-free-shadow-removal/blob/master/networks.py#L7-L14)?

**Comment by vinthony** [comment]
Other initializers might also work.
We follow the baseline of CAN in fast image processing: https://github.com/CQFIO/FastImageProcessing

**Comment by naoto0804** [comment]
Is it possible to share the masks that you used for the SRD dataset, which is essential for evaluation? I'm trying to extract it, but my result looks too noisy to be a ground truth;

**Comment by vinthony** [comment]
Hi, we also use a simple rule to extract the mask from the image pair.
Although it is not totally correct, we found it still helps for shadow removal.
Here is the link of the mask of SRD dataset. [OneDrive](https://uofmacau-my.sharepoint.com/:u:/g/personal/yb87432_umac_mo/EZ8CiIhNADlAkA4Fhim_QzgBfDeI7qdUrt6wv2EVxZSc2w?e=hZ0ruG)

**Comment by naoto0804** [comment]
I really really appreciate it;

**Comment by ruchu001** [comment]
链接不可用
